### DISCUSSION TITLE
use in flutter

### DISCUSSION THREAD

**Discussion by OmidHaqi** [pocketbase/pocketbase#4015] [Q&A] [answered]
Hi, 
I want to use pocketbase in a small project, but I can't request api. Can anyone help me?  
I will give you pictures of my code
Error in debug console:

```
ClientException (ClientException: {url: http://127.0.0.1:8090/api/collections/Article/records?page=1&perPage=50&filter=created+%3E%3D+%222022-01-
01+00%3A00%3A00%22+%26%26+someField1+%21%3D+someField2&skipTotal=false, isAbort: true, statusCode: 0, response: 
{}, originalError: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), 
address = 127.0.0.1, port = 49298, uri=http://127.0.0.1:8090/api/collections/Article/records?
page=1&perPage=50&filter=created+%3E%3D+%222022-01-
01+00%3A00%3A00%22+%26%26+someField1+%21%3D+someField2&skipTotal=false})
![Screenshot from 2023-12-23 13-06-50](https://github.com/pocketbase/pocketbase/assets/92772541/10c419c3-4792-420e-
945d-3f2540018d63)
```

**myPocketBase class** 

![Screenshot from 2023-12-23 13-08-27](https://github.com/pocketbase/pocketbase/assets/92772541/8369ef16-7f1e-4474-8e58-797a07481520)

**init State func** 

![Screenshot from 2023-12-23 13-10-22](https://github.com/pocketbase/pocketbase/assets/92772541/4b6cd45f-db16-4e6a-9f69-2844fb175fdd)

 **my model class** 

![Screenshot from 2023-12-23 13-07-51](https://github.com/pocketbase/pocketbase/assets/92772541/3698a756-7412-444c-b8e8-dbef59ae388b)

**debug console**

![Screenshot from 2023-12-23 13-06-50](https://github.com/pocketbase/pocketbase/assets/92772541/6016e19d-8115-4b4a-98f0-fa00d76328fc)

**Comment by ganigeorgiev** [comment]
How do you run the Flutter app and what platform do you target - mobile, desktop, web?

I'm having difficulties reading the screenshots, but based on the error message, I don't think this is PocketBase related and you'll probably stumble on the same error with any other service that you run on localhost and attempt to access it from a mobile emulator.

If you are testing with the Android emulator, instead of `http://127.0.0.1:8090` you can try with `http://10.0.2.2:8090` when initializing the PocketBase Dart SDK.
Alternatively, you can also use [ngrok](https://ngrok.com/download) (eg. `ngrok http 8090`), which will create a TLS public address for your local PocketBase server and you can use the generated address in the PocketBase Dart SDK initialization.

**Reply by OmidHaqi** [reply]
my target is mobile 

im use http://10.0.2.2:8090 but 
![Screenshot from 2023-12-23 13-30-24](https://github.com/pocketbase/pocketbase/assets/92772541/3b301cea-4e5c-4883-ad1c-6f4765b9da2d)

I proceeded with this [tutorial](https://youtu.be/smV5blNXQLg?si=IQW2TcorDXphSeye)

**Reply by ganigeorgiev** [reply, marked as answer]
_Please stop posting code and errors as screenshots. It is hard to read and prevent others finding your issue later when searching for similar problem._

The 400 error indicates that some your filters are probably incorrect (eg. missing fields or typos). For more details about the API errors you can either start the PocketBase executable with `--dev` and/or check the failed requests in the _Admin UI > Logs_.

**Reply by OmidHaqi** [reply]
_Sorry, I wasn't paying attention._

I checked the log and realized that I have an extra filter inside the code.
I deleted it and it was fixed. 
Thank you for your time.
